code/datums/components/omen.dm

```diff
@@ -116,3 +128,24 @@
 		return
 
 	qdel(src)
+
+/** Unfortunate quirk players delimb on death */
+/datum/component/omen/proc/check_death(mob/living/our_guy)
+	SIGNAL_HANDLER
+
+	if(!permanent)
+		return FALSE
+
+	var/turf/tile = get_turf(player)
+	if(tile)
+		explosion(tile,  devastation_range = explode_inner, heavy_impact_range = explode_inner, light_impact_range = explode_outer, flame_range = explode_inner, flash_range = explode_inner, explosion_cause = src)
+
+	if(!iscarbon(player))
+		our_guy.gib()
+		return TRUE
+
+	var/mob/living/carbon/player = our_guy
+	for(var/obj/item/bodypart/thing in user.bodyparts)
```

Comment:
I think the proc `our_guy.spread_bodyparts()` will do this for you, and you can also call `spill_organs` and `spawn_gibs` if you want to make it look even more like being gibbed without actually deleting the mob.

Reply:
Nice catch, I just lifted this from amputation shears ha